segno/encoder.py

```diff
@@ -1082,7 +1082,7 @@ def make_segment(data, mode, encoding=None):
         if _PY2:  # pragma: no cover
             segment_data = [ord(b) for b in segment_data]
         # Note: len(segment.data)! segment.data_length = len(segment.data) / 2!!
-        for i in range(0, char_count, 2):
```

Comment:
Here we should use `len(segment_data)` as upper bound. Otherwise only half data is added to segment.